### PR TITLE
Support Windows

### PR DESCRIPTION
**Pull request by daipom**
This adds Windows support.

I added default configuration values for Windows, and separated the parsing logic.

The command for Windows had to be complicated, 
but I made it possible to change the keys without setting the command in Windows
so that users don't have to understand Windows command.

### CODE REVIEW

**Comment by ashie** [comment]
@y-ken @cosmo0920 Could you kindly take a look this?

**Comment by y-ken** [comment]
Looks like so great work however I cannot verify on windows, could anybody check on windows 10?

**Comment by cosmo0920** [comment]
I'll take a look on this.

**Comment by daipom** [comment]
@y-ken @cosmo0920 Thank you very much!

File: README.md [hunk 1]
```diff
@@ -90,6 +96,38 @@ $ tail -f /var/log/td-agent/td-agent.log
   * to use short hostname, set `hostname -s` for this option on linux/mac.
   * [default] `hostname`
 
+### About Windows
+
+Default `command` in Windows below is complicated, but you can fix `keys` without fixing `command`, unless you specify a key other than the default.
```
Comment (y-ken) [inline README.md:101]: I guess this is more user-friendly English
```suggestion
Default `command` preset for Windows provides many of keys as below. Generally, you can pick up the columns with `keys` option.
If you need additional keys, consider to update `command` option
```

**Comment by y-ken** [comment]
If some windows user using Fluentd on Windows 8 , 7 or Vista or earlier version, How about to add a note as like "this is working after Windows *** or later"?
or, do we need to specify powershell version instead?

**Comment by cosmo0920** [comment]
> If some windows user using Fluentd on Windows 8 , 7 or Vista or earlier version, How about to add a note as like "this is working after Windows *** or later"?
> or, do we need to specify powershell version instead?

I think that we should describe which versions we used for confirming this feature on Windows.

e.g) Windows 10 w/ PowerShell 5.1 and Windows 10 w/ PowerShell 7.1 is confirmed to be working well.

**Comment by cosmo0920** [comment]
The reasons why I happened to find that this plugin w/ PowerShell 7 is not workable, I'm usually using PowerShell 7 is the default shell of Windows Terminal....

**Review by cosmo0920 — COMMENTED** [review]
I added a few commits to support newer PowerShell environment such as PowerShell 7.
Otherwise, it looks good to me.
Thanks for your work! :punch:

**Review by cosmo0920 — APPROVED** [review]

**Comment by cosmo0920** [comment]
I'd merged this PR into master.
@y-ken Could you release the new version of this plugin?
I think that the next version should be v0.2.0 or something.

**Comment by cosmo0920** [comment]
@ashie or @kenhys, Could you add using your confirmation environment information under https://github.com/y-ken/fluent-plugin-watch-process#about-windows?
I don't use Windows server platform to confirm this PR behavior. If you use such platform, please add their information there.

**Comment by y-ken** [comment]
I'm preparing to release v0.2.0.
And... Good news, I have added you in a owner list
image

**Comment by y-ken** [comment]
released v0.2.0

https://rubygems.org/gems/fluent-plugin-watch-process

**Comment by ashie** [comment]
> And... Good news, I have added you in a owner list

Oh! I've confirmed it. Thanks!

**Comment by daipom** [comment]
@y-ken @cosmo0920 @ashie 
Thank you for the review and fix! I'm so happy this fix was merged!
I did not check enough about the Powershell version. Thank you very much for the fix.

**Comment by cosmo0920** [comment]
Thanks. I'd received the ownership of this plugin. :punch:

**Pull request merged**